rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java

```diff
@@ -61,7 +63,7 @@ private void validateInternalRedeemScript(Script internalRedeemScript) {
             throw new RedeemScriptCreationException(message, INVALID_INTERNAL_REDEEM_SCRIPTS);
         }
 
-        if (redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.FLYOVER) {
+        if (redeemScriptParser instanceof FlyoverRedeemScriptParser) {
```

Comment:
I think we should get rid of this validation. Respect polymorphism and allow a flyover redeem script to be created with whatever script comes

Reply:
Done.

Also, tests were udpated.